code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -319,6 +319,21 @@
 		return
 	ui_interact(user)
 
+/obj/machinery/chem_dispenser/AltClick(mob/user)
+	if(!is_drink)
+		return
+	if(user.incapacitated())
+		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
+		return
+	if(!Adjacent(user))
+		return
+	if(anchored)
+		to_chat(user, "<span class='warning'>[src] is anchored to the floor!</span>")
+		return
```

Comment:
The `is_drink` and `Adjacent()` checks can be merged together to save a couple of lines if you want, since they're both just returning:
```suggestion
/obj/machinery/chem_dispenser/AltClick(mob/user)
	if(!is_drink || !Adjacent(user))
		return
	if(user.incapacitated())
		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
		return
	if(anchored)
		to_chat(user, "<span class='warning'>[src] is anchored to the floor!</span>")
		return
```
It also stops the "You can't do that right now!" message showing even if you're not adjacent to it, since that check will come first.

Reply:
While I see your points, I don't agree since that means regular chem dispensers will ALWAYS give this message.